### DISCUSSION TITLE
Jamulus Server Setup Problem Ubuntu 22.04

### DISCUSSION THREAD

**Discussion by lolhun1988** [jamulussoftware/jamulus#3253] [Technical Support & Help] [answered]
Dear all,

I have installed yesterday the latest Jamulus server. I struggle getting it running. I have followed all steps and seems to be running in the background but it is not registered to any genre.

This is my bash file currently.

```
[Unit]
Description=Jamulus headless server
After=network.target
StartLimitIntervalSec=0

[Service]
Type=simple
User=jamulus
Group=nogroup
NoNewPrivileges=true
ProtectSystem=true
ProtectHome=true
Nice=-20
IOSchedulingClass=realtime
IOSchedulingPriority=0

#### Change this to publish this server, set genre, location and other parameters.
#### See https://jamulus.io/wiki/Command-Line-Options ####
ExecStart=/bin/sh -c 'exec /usr/bin/jamulus-headless -s -n' --nogui --server \
--directoryaddress "anygenre1.jamulus.io:22124" \
--serverinfo "TEST999;Amsterdam;NL" \
--welcomemessage "Welcome Sucker!" \
--numchannels 20

Restart=on-failure
RestartSec=30
StandardOutput=journal
StandardError=inherit
SyslogIdentifier=jamulus


[Install]
WantedBy=multi-user.target
```

When checking with ps aux, it seems to be running.

jamulus     2439  0.0  0.2 170432 51752 ?        S<sl 10:18   0:00 /usr/bin/jamulus-headless -s -n

However cannot see it online, it is not a network issue because if I host one on my Mac it goes online immediately as it should.

Any tip or lead is welcome. Thank you.

**Comment by rdica** [comment]
Can you paste the output of
`ps wwwl $(pgrep jamulus)`

The ExecStart line in that systemd unit file looks suspect.

What version did you install, and from where did you get the install pkg?

**Reply by lolhun1988** [reply]
Here you go:

F   UID     PID    PPID PRI  NI    VSZ   RSS WCHAN  STAT TTY        TIME COMMAND
4   117    3756       1   0 -20 170436 51756 -      S<sl ?          0:00 /usr/bin/jamulus-headless -s -n

I have followed instructions on the website (link below) for headless since it is on a cloud. However I tried on a VM on my computer same result.

https://jamulus.io/wiki/Running-a-Server

Since I am not an experienced linux user for sure the syntax by now is not OK, tried without quotation mark as well as it was in the original file and the suggestion on the website but no luck. It seem to me the commands are not getting passed to jamulus to register it on anygenre and etc...

**Reply by rdica** [reply, marked as answer]
Your output should have looked similar to:
```
F   UID     PID    PPID PRI  NI    VSZ   RSS WCHAN  STAT TTY        TIME COMMAND
4  1002     609       1  20   0 318292 17516 do_pol Ssl  ?        153:41 /usr/bin/jamulus --server --nogui --discononquit --welcomemessage /jamulus/html/jam.html --numchannels 15 --directoryaddress anygenre2.jamulus.io:22224 --serverinfo server name;Manila;ph -R /jamulus/recordings --norecord -l /jamulus/logs/jamulus-server.log -T -P
```
Your output doesn't show the additional program arguments shown in the `ExecStart` line after `'exec /usr/bin/jamulus-headless -s -n'`. Since those arguments are not added to the command, there's no attempt to connect and register with a directory server.

You can try editing that file, adding a `#` in front of the existing `ExecStart` line, then below that type in a new line:
```
ExecStart = /usr/bin/jamulus-headless -s -n
```

When you are done editing, the section should look like:

```
#ExecStart=/bin/sh -c 'exec /usr/bin/jamulus-headless -s -n' --nogui --server \
ExecStart = /usr/bin/jamulus-headless -s -n \
--directoryaddress "anygenre1.jamulus.io:22124" \
--serverinfo "TEST999;Amsterdam;NL" \
--welcomemessage "Welcome Sucker!" \
--numchannels 20
```

Once you've saved the file, `sudo systemctl daemon-reload`, then restart the jamulus server. If your jamulus server is configured to start at bootup, you can just reboot without having to run the previous commands manually.

**Reply by rdica** [reply]
I forgot to mention, `-s -n` is the short form arguments for `--nogui --server`. Since they're the same there was no need to include them twice. :)

**Reply by lolhun1988** [reply]
Thank you I will try this tonight!

**Reply by ann0see** [reply]
I think you were also using an old version of the systemd service file. the /bin/sh -c should not be there at all (https://github.com/jamulussoftware/jamulus/pull/3187)

Edit: just checked and saw that the current stable version still has this. Seems to be soon time for a release.

**Reply by lolhun1988** [reply]
It can be because the cloud is running on ARM and my Macbook with the VM is also ARM. This is what the website instruction got me.

**Reply by ann0see** [reply]
No. I believe the confusion is there due to the fact that the mentioned pull request isn't yet in the stable release. ARM and amd64 (Intel) should work similar.

**Reply by rdica** [reply]
@ann0see I had to doublecheck that. I know my PR got approved, I just couldn't remember if that change had been released yet, research showed I committed three days after the last release heh
